utilities/roswtf/src/roswtf/network.py

```diff
@@ -49,7 +49,7 @@ def ip_check(ctx):
     local_addrs = rosgraph.network.get_local_addresses()
 
     resolved_ips = [host[4][0] for host in socket.getaddrinfo(socket.gethostname(), 0, 0, 0, socket.SOL_TCP)]
-    global_ips = [ ip for ip in resolved_ips if not ip.startswith('127.') and not ip == '::1']
+    global_ips = [ ip.replace('%', '%%') for ip in resolved_ips if not ip.startswith('127.') and not ip == '::1']
```

Comment:
I would also argue that the returned message should not contain any escaping yet since it should be agnostic to how it is used on the outside. Instead the escaping should happen right before the replacement is currently failing.